src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,170 @@
+    private Node<K, V>[] table;
+    private int size;
+    private final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int threshold;
+    private boolean isResized = false;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        threshold = (int) (DEFAULT_LOAD_FACTOR * table.length);
+
+        if (size >= threshold) {
+            resize();
+        }
+
+        int index = (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+        Node<K, V> nodeToPut;
+        if (index == 0) {
+            nodeToPut = new Node<>(index, key, value, null);
+        } else {
+            nodeToPut = new Node<>(key.hashCode(), key, value, null);
+        }
+
+        if (table[index] == null) {
+            table[index] = nodeToPut;
+            size++;
+            return;
+        }
+
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+
+            if (currentNode.next == null) {
+                currentNode.next = nodeToPut;
+                size++;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
 
     }
 
     @Override
     public V getValue(K key) {
+        int index = (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+        Node<K, V> currentNode = table[index];
+
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj.getClass() != getClass()) {
+            return false;
+        }
+
+        MyHashMap<K, V> other = (MyHashMap<K, V>) obj;
+
+        if (size != other.size) {
+            return false;
+        }
+
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                Object otherValue = other.getValue(node.key);
+                if (!Objects.equals(node.value, otherValue)) {
+                    return false;
+                }
+                node = node.next;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = 0;
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                hash += Objects.hashCode(node.key) ^ Objects.hashCode(node.value);
+                node = node.next;
+            }
+        }
+        return hash;
+    }
+
+    private void resize() {
```

Comment:
The `resize` method should update the `size` variable to reflect the actual number of elements in the new table. Currently, it only updates the `table` and `threshold` but not the `size`.